docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+
+### Connection Management
+
+ANTA uses the `httpx` library as its underlying HTTP client for device connections. Each device has his own connection pool with a default maximum of `100` connections. On large fabrics, this can lead to a very high number of connections (n x 100), which may overwhelm the ANTA process. Connection pooling can be tuned via:
+
+```bash
+# Maximum number of allowable connections
+export ANTA_MAX_CONNECTIONS=100
+
+# Number of allowable keep-alive connections
+export ANTA_MAX_KEEPALIVE_CONNECTIONS=20
+
+# Time limit on idle keep-alive connections in seconds
+export ANTA_KEEPALIVE_EXPIRY=5.0
+```
+
+**Best Practice**: For large fabrics, limiting connections (5-10 per device) has shown optimal performance in testing. `ANTA_MAX_KEEPALIVE_CONNECTIONS` should be lower than `ANTA_MAX_CONNECTIONS`. See [HTTPX documentation](https://www.python-httpx.org/advanced/resource-limits/) for more details.
+
+### Timeouts Configuration
+
+ANTA provides several environment variables to control `httpx` timeouts:
+
+```bash
+# Global timeout
+export ANTA_TIMEOUT=30.0
+
+# Maximum amount of time to wait until a socket connection to the requested host is established
+export ANTA_CONNECT_TIMEOUT=30.0
+
+# Maximum duration to wait for a chunk of data to be received (for example, a chunk of the response body)
+export ANTA_READ_TIMEOUT=30.0
+
+# Maximum duration to wait for a chunk of data to be sent (for example, a chunk of the request body)
+export ANTA_WRITE_TIMEOUT=30.0
+
+# Maximum duration to wait for acquiring a connection from the connection pool
+export ANTA_POOL_TIMEOUT=30.0
+```
+
+If not set, these values default to the global `timeout` of ANTA, which is `30.0` seconds by default. The global timeout can also be set via the `--timeout` command-line option. See [HTTPX documentation](https://www.python-httpx.org/advanced/timeouts/) for more details.
+
+### Performance Tuning
+
+For optimal single-process performance, consider the following tuning parameters:
+
+1. Adjust concurrency based on test count and system resources:
+
+    ```bash
+    # For 250 devices with 100 tests each (25,000 total tests)
+    export ANTA_MAX_CONCURRENCY=25000
+    ```
+
+    Even though the default value is `10000`, it may not be optimal for your system. You can increase or decrease this value gradually to find the best setting for your environment.
+
+2. Optimize connection pooling:
+
+    ```bash
+    export ANTA_MAX_CONNECTIONS=10
+    export ANTA_MAX_KEEPALIVE_CONNECTIONS=5
+    ```
+
+    These values are a good starting point for large fabrics. Adjust them based on your fabric size and performance testing.
+
+3. Increase timeouts for stability:
+
+    ```bash
+    export ANTA_TIMEOUT=3600.0
+    ```
+
+    This is **very** important for large fabrics. Increase the global timeout to prevent test failures due to timeouts. You shouldn't need to adjust the other timeout values unless you have specific requirements.
```

Comment:
Hwo does this manifest?
I feel very important should be all bold and potentially this should be number 1 in the list :)